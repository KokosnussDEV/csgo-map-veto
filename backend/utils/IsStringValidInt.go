package utils

import (
	"strconv"
)

func IsStringValidInt(value string) bool {
	if _, err := strconv.Atoi(value); err == nil {
		return true
	}

	return false
}
