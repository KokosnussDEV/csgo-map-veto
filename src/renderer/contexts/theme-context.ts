import { createContext } from 'react'

type Theme = {
  dark: '#222' | '#fff'
  light: '#333' | '#f5f5f5'
  darkInversed: '#222' | '#fff'
  lightInversed: '#333' | '#f5f5f5'
  primary: '#1976d2' | '#90caf9'
  danger: '#f56565'
  success: '#48bb78'
  warning: '#ed8936'
}

const THEME_DARK = 'dark'
const THEME_LIGHT = 'light'
type ThemesKeys = typeof THEME_DARK | typeof THEME_LIGHT

type Themes = {
  [THEME_DARK]: Theme
  [THEME_LIGHT]: Theme
}

const themes: Themes = {
  [THEME_LIGHT]: {
    dark: '#fff',
    darkInversed: '#222',
    light: '#f5f5f5',
    lightInversed: '#333',
    primary: '#1976d2',
    danger: '#f56565',
    success: '#48bb78',
    warning: '#ed8936',
  },
  [THEME_DARK]: {
    dark: '#222',
    darkInversed: '#fff',
    light: '#333',
    lightInversed: '#f5f5f5',
    primary: '#90caf9',
    danger: '#f56565',
    success: '#48bb78',
    warning: '#ed8936',
  },
}

type ThemeContext = {
  theme: Theme
  toggleTheme: () => void
}

const AppThemeContext = createContext<ThemeContext>({
  theme: themes[THEME_DARK],
  toggleTheme: () => {}, // eslint-disable-line
})

export { Theme, THEME_DARK, THEME_LIGHT, ThemesKeys, AppThemeContext, themes }
