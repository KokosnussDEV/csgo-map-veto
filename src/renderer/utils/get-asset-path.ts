import path from 'path'

/**
 * Return the correct path to a static asset from the "static" folder.
 * When running in dev mode, webpack dev server try to get assets from webpack-dev-server:
 * http://localhost:9000/static/images/img.png
 * But the correct "path" is:
 * http://localhost:9000/images/img.png
 */
const getAssetPath = (assetPath: string) => {
  if (process.env.NODE_ENV !== 'production') {
    return `./${assetPath}`
  }

  return path.join(__static, assetPath)
}

export { getAssetPath }
