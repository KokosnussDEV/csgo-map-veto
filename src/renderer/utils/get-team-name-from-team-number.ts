import { TeamNumbers, TeamNumber } from 'renderer/types/team-number'

const getTeamNameByTeamNumber = (teamNumber: TeamNumber, teamOneName: string, teamTwoName: string) => {
  switch (teamNumber) {
    case TeamNumbers.TEAM1:
      return teamOneName
    case TeamNumbers.TEAM2:
      return teamTwoName
    default:
      return 'SERVER'
  }
}

export { getTeamNameByTeamNumber }
