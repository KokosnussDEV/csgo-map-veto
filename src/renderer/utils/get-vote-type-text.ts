import { VoteTypes, VoteType } from 'renderer/types/vote-type'

const getVoteTypeText = (voteType: VoteType) => {
  return voteType === VoteTypes.BAN ? 'banned' : 'picked'
}

export { getVoteTypeText }
