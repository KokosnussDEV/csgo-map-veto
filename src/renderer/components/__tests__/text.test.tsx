import React from 'react'
import { renderWithTheme } from 'test/utils'
import { themes } from 'renderer/contexts/theme-context'
import { Text } from '../text'

describe('Text', () => {
  it('should render text', () => {
    const { getByText } = renderWithTheme(<Text>A text</Text>)

    expect(getByText('A text')).toBeTruthy()
  })

  it('should render with dark theme', () => {
    const { container } = renderWithTheme(<Text>A text</Text>, themes.dark)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render with light theme', () => {
    const { container } = renderWithTheme(<Text>A text</Text>, themes.light)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render with custom styles properties', () => {
    const { container } = renderWithTheme(
      <Text isBold={true} isUppercase={true} textAlign="center" size="xl">
        A text
      </Text>,
      themes.light
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
