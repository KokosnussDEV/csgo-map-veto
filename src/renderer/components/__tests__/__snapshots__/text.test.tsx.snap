// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render with custom styles properties 1`] = `
.c0 {
  text-transform: uppercase;
  font-family: Lato-bold;
  text-align: center;
  font-size: 22px;
  color: #222;
}

<p
  class="c0"
>
  A text
</p>
`;

exports[`Text should render with dark theme 1`] = `
.c0 {
  font-size: 16px;
  color: #fff;
}

<p
  class="c0"
>
  A text
</p>
`;

exports[`Text should render with light theme 1`] = `
.c0 {
  font-size: 16px;
  color: #222;
}

<p
  class="c0"
>
  A text
</p>
`;
